src/Core/Repository/AbstractMultiShopObjectModelRepository.php

```diff
@@ -173,4 +174,42 @@ protected function checkShopAssociation(int $id, string $objectModelClassName, S
             ));
         }
     }
+
+    /**
+     * @param ObjectModel $objectModel
+     * @param ShopId[]|int[] $shopIds
+     * @param string $exceptionClass
+     * @param int $errorCode
+     *
+     * @throws CoreException
+     */
+    protected function deleteObjectModelFromShops(ObjectModel $objectModel, array $shopIds, string $exceptionClass, int $errorCode = 0): void
+    {
+        try {
+            $intShopIds = array_map(function ($shopId) {
+                return $shopId instanceof ShopId ? $shopId->getValue() : (int) $shopId;
+            }, $shopIds);
+
+            // Use this field to indicate which shop associations need to be removed
+            $objectModel->id_shop_list = $intShopIds;
```

Comment:
If im not mistaken the $intShopIds var is not used anywhere else.

```suggestion
            $objectModel->id_shop_list = array_map(function ($shopId): int {
                return $shopId instanceof ShopId ? $shopId->getValue() : (int) $shopId;
            }, $shopIds);
```

Reply:
Yeah it was mostly to ease debug, but the variable is not necessary I'll remove it